README.md

```diff
@@ -24,11 +24,183 @@ echo $markdown;
 // Hello world!
 ```
 
-## Contributing
+## Contributing
 
 See the [Contributing] document for guidance on making contributions to the
 project.
 
+## API
+
+Classes in this library are namespaced in `CopilotTags` (e.g. `CopilotTags\Paragraph`).
+
+### CopilotTag
+
+Interface for tag generator classes.
+* `CopilotTag->write()`
+
+  Write tag object as beautified Copilot-flavored Markdown string.<br>
+  **Return:** string (Markdown)
+
+### Text
+
+Generator for unformatted text. The given text value can contain any valid
+Copilot-flavored Markdown.
+
+```php
+(new Text("Hello world!"))->write();
+// "Hello world!"
+```
+
+* `new Text($text)`<br>
+  ***text:*** string (Markdown)<br>
+
+### Heading
+
+Generator for [ATX headings](http://spec.commonmark.org/0.27/#atx-headings).
+
+```php
+(new Heading("Hello world!", 3))->write();
+// "### Hello world!\n"
+```
+
+* `new Heading($text[, $level])`<br>
+***text:*** string (Markdown)<br>
+***level:*** int (default: `2`)
+
+### Paragraph
+
+Generator for [paragraphs](http://spec.commonmark.org/0.27/#paragraphs).
+
+```php
+(new Paragraph("Hello world!"))->write();
+// "Hello world!\n\n"
+```
+
+* `new Paragraph($text)`<br>
+  ***text:*** string (Markdown)<br>
+
+### InlineText
+
+Generator for inline text tags: [emphasis](https://github.com/conde-nast-international/copilot-markdown/blob/master/specification/0E.md#3111-emphasis), [strong](http://spec.commonmark.org/0.27/#emphasis-and-strong-emphasis), [subscript](https://github.com/conde-nast-international/copilot-markdown/blob/master/specification/0E.md#3110-subscript), [superscript](https://github.com/conde-nast-international/copilot-markdown/blob/master/specification/0E.md#319-superscript) and
+[delete](https://github.com/conde-nast-international/copilot-markdown/blob/master/specification/0E.md#314-delete).
+
+```php
+(new InlineText("Hello world!", InlineTextDelimiter::EMPHASIS))->write();
+// "*Hello world!*"
+```
+
+* `new InlineText($text[, $delimiter])`<br>
+  ***text:*** string (Markdown)<br>
+  ***delimiter:*** string (`InlineTextDelimiter`) (default: `""`)
+
+#### InlineTextDelimiter
+
+|Class Constant |Value |Also known as              |
+|---------------|------|---------------------------|
+|`EMPHASIS`     |`*`   |Italic, em                 |
+|`STRONG`       |`**`  |Bold                       |
+|`SUBSCRIPT`    |`~`   |Inferior, sub              |
+|`SUPERSCRIPT`  |`^`   |Superior, super            |
+|`DELETE`       |`~~`  |Strikethrough, Strike, Del |
+
+### Link
+
+Generator for [links](https://github.com/conde-nast-international/copilot-markdown/blob/master/specification/0E.md#317-link).
+
+```php
+(new Link("Hello world!", "https://github.com/"))->write();
+// "[Hello world!](https://github.com/)"
+(new Link("Hello world!", "https://github.com/", array("foo"=>"bar")))->write();
+// "[Hello world!](https://github.com/){: foo=\"bar\" }"
```

Comment:
Is it worth using a real attribute here for clarity? like `_target=blank`

Reply:
@graycodes I thought about this but decided not to in case it was seen as an endorsement of that particular attribute. Do you know of an attribute that we would definitely want to encourage use of? I think `"target"="_blank"` is maybe not something we want to encourage.

Reply:
Well it's up to them as to what links to open in new contexts. I know they link to external sites quite often and they'd often want these to be in a new context. That said you could use something like `rel="nofollow"`.